### PR TITLE
add ros2-cartographer-slam page

### PR DESCRIPTION
**Pull request by pedro-fuoco**
Continuation of #5359.
The technical details have been heavily based on the  `ardupilot_ros` README, which is linked in the page

@Hwurzburg could you upload the video to Ardupilot's youtube channel ?

### CODE REVIEW

File: dev/source/docs/ros2-cartographer-slam.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
+.. _ros2-cartographer-slam:
+
+========================================
+Cartographer SLAM with ROS 2 in SITL
+========================================
```
Comment (Hwurzburg) [inline dev/source/docs/ros2-cartographer-slam.rst:5]: under/overlines wrong

File: dev/source/docs/ros2-cartographer-slam.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+This page shows how to setup ROS 2 with Ardupilot SITL and run Google Cartographer as a SLAM source. 
+
+Installation
+------------
```
Comment (Hwurzburg) [inline dev/source/docs/ros2-cartographer-slam.rst:10]: should use =====  for next heading level not ----

File: dev/source/docs/ros2-cartographer-slam.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+Now source your workspace and build `ardupilot_ros`:
+
+.. code-block:: bash
+
+    cd ~/ros2_ws
+    source ./install/setup.bash
+    colcon build --packages-select ardupilot_ros
+
+Usage
+-----
```
Comment (Hwurzburg) [inline dev/source/docs/ros2-cartographer-slam.rst:36]: =====

File: dev/source/docs/ros2-cartographer-slam.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+To test and fly around, you can launch a `mavproxy <https://ardupilot.org/dev/docs/copter-sitl-mavproxy-tutorial.html>`__ instance in another terminal:
+
+.. code-block:: bash
+    
+    mavproxy.py --console --map --aircraft test --master=:14550
+
+
+Configure ArduPilot
+-------------------
```
Comment (Hwurzburg) [inline dev/source/docs/ros2-cartographer-slam.rst:60]: ========

File: dev/source/docs/ros2-cartographer-slam.rst [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+-  :ref:`EK3_SRC1_VELZ <copter:EK3_SRC1_VELZ>` = 6 to set vertical velocity source to ExternalNAV
+-  :ref:`EK3_SRC1_YAW <copter:EK3_SRC1_YAW>` = 6 to set yaw source to ExternalNAV
+-  :ref:`VISO_TYPE <copter:VISO_TYPE>` = 1 to enable visual odometry
+-  :ref:`ARMING_CHECK <copter:ARMING_CHECK>` = 388598 (optional, to disable GPS checks)
+
+
+After changing the values above, reboot the flight controller.
+
+Demo
+----
```
Comment (Hwurzburg) [inline dev/source/docs/ros2-cartographer-slam.rst:79]: use =====

File: dev/source/docs/ros2-cartographer-slam.rst [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+Here's a quick video showcasing the results expected from this tutorial:
+
+..  youtube:: apZBxC2RHTM
```
Comment (Hwurzburg) [inline dev/source/docs/ros2-cartographer-slam.rst:83]:
```suggestion
..  youtube:: bpjGyAECKHA
```
new link

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
thanks...a few nits

**Comment by Hwurzburg** [comment]
thanks

**Comment by pedro-fuoco** [comment]
@tridge also asked me to link the common-ekf-sources page

https://github.com/ArduPilot/ardupilot_ros/pull/12#issuecomment-1696511247

I will submit a fixup

**Comment by pedro-fuoco** [comment]
@Hwurzburg Done! Let me know if I need to change anything else, thanks for the review

**Comment by Hwurzburg** [comment]
Question, how would you use this on a real vehicle per the warning? via companion computer?

**Comment by pedro-fuoco** [comment]
> Question, how would you use this on a real vehicle per the warning? via companion computer?

Yes, but the tutorials on how to setup ardupilot and ROS 2 with a companion computer are still not available. I suggest leaving the warning there preemptively since it is a safety measure.

**Comment by pedro-fuoco** [comment]
Removed the warning about rosdep since it is now supported https://github.com/ArduPilot/ardupilot_ros/commit/ca804c8b5952a65003675e3b5b30ad116d134753

**Comment by Hwurzburg** [comment]
LGTM...merging

**Pull request merged**